packages/contracts-bedrock/src/L1/OPContractsManager.sol

```diff
@@ -668,263 +668,77 @@ contract OPContractsManagerUpgrader is OPContractsManagerBase {
-                        payable(
-                            deployProxy({
-                                _l2ChainId: l2ChainId,
-                                _proxyAdmin: _opChainConfigs[i].proxyAdmin,
-                                _saltMixer: reusableSaltMixer(_opChainConfigs[i]),
-                                _contractName: "PermissionlessDelayedWETH-U16a"
-                            })
-                        )
-                    );
-
-                    // Initialize the DelayedWETH.
-                    upgradeToAndCall(
-                        _opChainConfigs[i].proxyAdmin,
-                        address(permissionlessDelayedWeth),
-                        impls.delayedWETHImpl,
-                        abi.encodeCall(IDelayedWETH.initialize, (_opChainConfigs[i].systemConfigProxy))
-                    );
-
-                    // Fix for stack too deep
-                    OPContractsManager.OpChainConfig memory opChainConfig = _opChainConfigs[i];
-                    // Deploy and set a new permissionless game to update its prestate
-                    deployAndSetNewGameImpl({
-                        _l2ChainId: l2ChainId,
-                        _disputeGame: IDisputeGame(address(permissionlessDisputeGame)),
-                        _newDelayedWeth: permissionlessDelayedWeth,
-                        _newAnchorStateRegistryProxy: anchorStateRegistry,
-                        _gameType: GameTypes.CANNON,
-                        _opChainConfig: opChainConfig
-                    });
-                }
-            }
+            // Do the chain upgrade.
+            // All of your updates should be done in this internal function unless you're making a
+            // change to how upgrades work in general.
+            _doChainUpgrade(impls, _opChainConfigs[i], l2ChainId);
 
             // Emit the upgraded event with the address of the caller. Since this will be a delegatecall,
             // the caller will be the value of the ADDRESS opcode.
             emit Upgraded(l2ChainId, _opChainConfigs[i].systemConfigProxy, address(this));
         }
     }
 
+    /// @notice Performs an upgrade for a specific chain.
+    /// @param _impls The implementations of the contracts.
+    /// @param _opChainConfig The configuration of the chain to upgrade.
+    /// @param _l2ChainId The L2 chain ID of the chain to upgrade.
+    function _doChainUpgrade(
+        OPContractsManager.Implementations memory _impls,
+        OPContractsManager.OpChainConfig memory _opChainConfig,
+        uint256 _l2ChainId
+    )
+        internal
+    {
+        // Use the SystemConfig to grab the DisputeGameFactory address.
+        IDisputeGameFactory dgf = IDisputeGameFactory(_opChainConfig.systemConfigProxy.disputeGameFactory());
+
+        // Need to upgrade the DisputeGameFactory implementation, no internal upgrade call.
+        upgradeTo(_opChainConfig.proxyAdmin, address(dgf), _impls.disputeGameFactoryImpl);
```

Comment:
why is this the only contract whose impl is being updated?

Reply:
A few more need to be updated too... more commits coming